exchanges/coinbasepro/coinbasepro_test.go

```diff
@@ -2,1042 +2,2062 @@ package coinbasepro
+	assert.ErrorIs(t, err, order.ErrAmountIsInvalid)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c, canManipulateRealOrders)
+	skipTestIfLowOnFunds(t)
+	id, err := uuid.NewV4()
+	assert.NoError(t, err)
+	resp, err := c.PlaceOrder(context.Background(), id.String(), testPair.String(), order.Sell.String(), "", order.Limit.String(), "", "CROSS", "", testAmount, testPrice, 0, 9999, false, time.Now().Add(time.Hour))
+	if assert.NoError(t, err) {
+		assert.NotEmpty(t, resp, errExpectedNonEmpty)
+	}
+	id, err = uuid.NewV4()
+	assert.NoError(t, err)
+	resp, err = c.PlaceOrder(context.Background(), id.String(), testPair.String(), order.Sell.String(), "", order.Limit.String(), "", "MULTI", "", testAmount, testPrice, 0, 9999, false, time.Now().Add(time.Hour))
+	require.NoError(t, err)
+	assert.NotEmpty(t, resp, errExpectedNonEmpty)
 }
 
-// TestGetFeeByTypeOfflineTradeFee logic test
-func TestGetFeeByTypeOfflineTradeFee(t *testing.T) {
-	feeBuilder := setFeeBuilder()
-	_, err := c.GetFeeByType(context.Background(), feeBuilder)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if !sharedtestvalues.AreAPICredentialsSet(c) {
-		if feeBuilder.FeeType != exchange.OfflineTradeFee {
-			t.Errorf("Expected %v, received %v", exchange.OfflineTradeFee, feeBuilder.FeeType)
-		}
-	} else {
-		if feeBuilder.FeeType != exchange.CryptocurrencyTradeFee {
-			t.Errorf("Expected %v, received %v", exchange.CryptocurrencyTradeFee, feeBuilder.FeeType)
+func orderTestHelper(t *testing.T, orderSide string) *GetAllOrdersResp {
+	t.Helper()
+	ordIDs, err := c.GetAllOrders(context.Background(), "", "", "", orderSide, "", "", "", "", "", []string{}, []string{}, 1000, time.Time{}, time.Time{})
+	assert.NoError(t, err)
+	if ordIDs == nil || len(ordIDs.Orders) == 0 {
+		t.Skip(skipInsufficientOrders)
+	}
+	for i := range ordIDs.Orders {
+		if ordIDs.Orders[i].Status == order.Open.String() {
+			ordIDs.Orders = ordIDs.Orders[i : i+1]
+			return ordIDs
 		}
 	}
+	t.Skip(skipInsufficientOrders)
+	return nil
 }
 
-func TestGetFee(t *testing.T) {
-	feeBuilder := setFeeBuilder()
-
-	if sharedtestvalues.AreAPICredentialsSet(c) {
-		// CryptocurrencyTradeFee Basic
-		if _, err := c.GetFee(context.Background(), feeBuilder); err != nil {
-			t.Error(err)
-		}
-
-		// CryptocurrencyTradeFee High quantity
-		feeBuilder = setFeeBuilder()
-		feeBuilder.Amount = 1000
-		feeBuilder.PurchasePrice = 1000
-		if _, err := c.GetFee(context.Background(), feeBuilder); err != nil {
-			t.Error(err)
-		}
-
-		// CryptocurrencyTradeFee IsMaker
-		feeBuilder = setFeeBuilder()
-		feeBuilder.IsMaker = true
-		if _, err := c.GetFee(context.Background(), feeBuilder); err != nil {
-			t.Error(err)
-		}
+func TestCancelOrders(t *testing.T) {
+	t.Parallel()
+	var orderSlice []string
+	_, err := c.CancelOrders(context.Background(), orderSlice)
+	assert.ErrorIs(t, err, errOrderIDEmpty)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c, canManipulateRealOrders)
+	ordIDs := orderTestHelper(t, "")
```

Comment:
This will auto cancel orders not specifically set, or fail if no orders are available. I suggest removing `orderTestHelper` and devs can input their bespoke order ID's in themselves. we shouldn't really be in the habit of trading on their behalf in test, even though they run with `canManipulateRealOrders`